pkg/controller/etcdproxy/certs.go

```diff
@@ -0,0 +1,252 @@
+			Create(newConfigMap(etcdstorage, etcdProxyCAConfigMapName(etcdstorage), c.config.ControllerNamespace, data))
+		if err != nil {
+			// TODO: refactor event handling (hint: see #40).
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+			return err
+		}
+		return nil
+	}
+	if err != nil {
+		// TODO: refactor event handling (hint: see #40).
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+	return nil
+}
+
+// createEtcdProxyServingCertSecret creates Secret in controller namespace with Etcd Proxy serving certificate and key.
+func (c EtcdProxyController) createEtcdProxyServingCertSecret(etcdstorage *etcdstoragev1alpha1.EtcdStorage,
+	serverCert, serverKey []byte) error {
+	// Secret for the etcd proxy server certs in controller namespace.
+	_, err := c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).
+		Get(etcdProxyServerCertsSecret(etcdstorage), metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		data := map[string][]byte{
+			"tls.crt": serverCert,
+			"tls.key": serverKey,
+		}
+		_, err = c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).
+			Create(newSecret(etcdstorage, etcdProxyServerCertsSecret(etcdstorage), c.config.ControllerNamespace, data))
+		if err != nil {
+			// TODO: refactor event handling (hint: see #40).
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+			return err
+		}
+		return nil
+	}
+	if err != nil {
+		// TODO: refactor event handling (hint: see #40).
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+	return nil
+}
+
+// updateAPIServerServingCAConfigMaps updates the ConfigMap in the aggregated API server namespace with the CA certificate.
+func (c *EtcdProxyController) updateAPIServerServingCAConfigMaps(etcdstorage *etcdstoragev1alpha1.EtcdStorage,
+	serverSignerCert []byte) error {
+	var errs []error
+	// Check are ConfigMap name and namespace provided.
+	for _, configMap := range etcdstorage.Spec.CACertConfigMaps {
+		caConfigMap, err := c.kubeclientset.CoreV1().ConfigMaps(configMap.Namespace).
+			Get(configMap.Name, metav1.GetOptions{})
+		if err != nil {
+			// TODO: refactor event handling (hint: see #40).
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+			errs = append(errs, err)
+			continue
+		}
+
+		caConfigMapCopy := caConfigMap.DeepCopy()
+		caConfigMapCopy.Data = map[string]string{"server-ca.crt": string(serverSignerCert)}
+
+		if val, ok := caConfigMapCopy.Annotations["etcd.xmudrii.com/certificate-generated"]; ok == true {
+			if val == "true" {
+				return nil
+			}
+		}
+
+		// TODO: extend annotations to include more information about certs, including expiry date, etc.
+		// HINT: take a look at openshift/service-serving-cert-signer for ideas.
+		caConfigMapCopy.Annotations = map[string]string{
+			"etcd.xmudrii.com/certificate-generated": "true",
```

Comment:
move annotation key to const.

Reply:
Done. Let me know is the const name and location appropriate.